astropy/io/votable/tree.py

```diff
@@ -2737,10 +2737,20 @@ def parse(self, iterator, config):
                     raise ValueError("Some specified column numbers out of range")
                 colnumbers = columns
             elif issubclass(columns.dtype.type, np.character):
-                try:
-                    colnumbers = [names.index(x) for x in columns]
-                except ValueError:
-                    raise ValueError(f"Columns '{columns}' not found in fields list")
```

Comment:
An alternative would be something like
```python
                    raise ValueError(f"Columns '{set(columns) - set(names)}' not found in fields list")
```
We don't really have to worry about performance too much if the code is only executed in case of an error.

Reply:
Agreed that performance doesn't matter here, however I'd prefer to preserve the order in which columns were specified, so set operations are not working here in my opinion. However, I still refactored my first patch to be much more condensed and hopefully clearer.

Reply:
I can see that maintaining the order of missing column names would be useful (e.g. for testing the error message), but I'd rather not worry about the listing them at all unless it is known that it is needed. The following will ensure that the missing columns will be in a `list` while delaying listing them as much as possible:
```python
                    raise ValueError(f"Columns '{[c for c in columns if c not in names]}' not found in fields list")
```

Reply:
even better. Thanks ! I also extended the unit test while I was at it.